File: packages/js-dpp/lib/errors/DataContractAlreadyPresentError.js
const ConsensusError = require('./ConsensusError');

class DataContractAlreadyPresentError extends ConsensusError {
  /**
   * @param {DataContract} dataContract
   */
  constructor(dataContract) {
    super('Data Contract is already present');

    this.dataContract = dataContract;
  }

  /**
   * Get Data Contract
   *
   * @return {DataContract}
   */
  getDataContract() {
    return this.dataContract;
  }
}

module.exports = DataContractAlreadyPresentError;

File: packages/rs-dpp/src/errors/consensus/state/data_contract/data_contract_already_present_error.rs
use crate::errors::consensus::state::state_error::StateError;
use crate::errors::consensus::ConsensusError;
use crate::errors::ProtocolError;
use bincode::{Decode, Encode};
use platform_serialization_derive::{PlatformDeserialize, PlatformSerialize};
use platform_value::Identifier;
use thiserror::Error;

#[derive(
    Error, Debug, Clone, PartialEq, Eq, Encode, Decode, PlatformSerialize, PlatformDeserialize,
)]
#[error("Data Contract {data_contract_id} is already present")]
#[platform_serialize(unversioned)]
#[ferment_macro::export]
pub struct DataContractAlreadyPresentError {
    /*

    DO NOT CHANGE ORDER OF FIELDS WITHOUT INTRODUCING OF NEW VERSION

    */
    pub data_contract_id: Identifier,
}

impl DataContractAlreadyPresentError {
    pub fn new(data_contract_id: Identifier) -> Self {
        Self { data_contract_id }
    }

    pub fn data_contract_id(&self) -> &Identifier {
        &self.data_contract_id
    }
}

impl From<DataContractAlreadyPresentError> for ConsensusError {
    fn from(err: DataContractAlreadyPresentError) -> Self {
        Self::StateError(StateError::DataContractAlreadyPresentError(err))
    }
}
